src/main/scala/models/Opportunity.scala

```diff
@@ -26,7 +26,21 @@ case class Opportunity(
                         duplicatedFrom: Option[OpportunityId],
                         description: Seq[OpportunityDescriptionSection]
                       ) {
+
   lazy val summary: OpportunitySummary = OpportunitySummary(id, title, startDate, endDate, value, publishedAt, duplicatedFrom)
+
+  lazy val statusString: String = {
+    publishedAt.isDefined match {
+      case true if isEndDatePassed => "Closed"
+      case true if isStartDatePassed => "Open"
+      case true => "Queued"
+      case false => "Draft"
+    }
+  }
+
+  lazy val isStartDatePassed = startDate.isEqual(LocalDate.now) || startDate.isBefore(LocalDate.now)
```

Comment:
AIUI, This is actually quite tricky, as in the unit test `testDateCurrent` you define the startdate for now, and rely on execution not taking longer than 1 millisecond before testing against isStartDatePassed which, I think, is new'ing up a LocalDate.  I may be wrong here though.

In practise I couldn't make it go wrong even when making my machine compile a bunch of updates :) so perhaps scala is so quick you don't worry about such things.